lang/rust-bootstrap/Portfile

```diff
@@ -269,186 +297,228 @@ livecheck.url               https://github.com/rust-lang/rust/tags
 livecheck.regex             refs/tags/(\[\\d\\.\]+).zip
 livecheck.version           ${version_current}
 
-# The root cause of offline mode is prevent cargo to check signatures
-# which is broken before 10.8
-# See: https://github.com/rust-lang/cargo/issues/4813
-# See: https://github.com/mozilla/releases-comm-central/commit/5cd3595203de78a6f47324f53f0cf616f4e383ea
-
-# use cargo2port to update it
-# cargo from rust bootstrap is linked against the system's crypto library
-# and old systems may not support the required SSL/TLS version to fetch anything from github for example
+# use cargo2port to update cargo.crates
+# cargo2port might have to be run in multiple directories
+#     ${worksrcpath}
+#     ${worksrcpath}/src/bootstrap
+#     ${worksrcpath}/src/tools/rust-analyzer
+#     ${worksrcpath}/??? for testing
+#     ${worksrcpath}/src/tools/cargo (for rust-bootstrap port)
+#     ${worksrcpath}/compiler/rustc_codegen_gcc (for https://github.com/rust-lang/rustc_codegen_gcc)
```

Comment:
Good news, I have a PR for `cargo2port` that will allow parsing multiple `Cargo.lock` files. We should identify those as required and sort this down since it requires a bit of a complex set of text parsing until this is merged and released.

Reply:
That is wonderful news!
As I am sure you know, having to use `cat` and `sort -u` on the output of `cargo2port` is a bit of a pain.

Please forgive me, but I am afraid I am not quite sure what is meant by 
> We should identify those as required and sort this down ...

Reply:
I think I was flagging by time I got to that particular comment, so my wording was obviously not as clear as I wanted. What I mean here is that your comment is a bit ambiguous as to *which* Cargo.lock files should be used for updating these portfiles. Ideally, we should identify *only* the ones needed and put the comment required to provide a new `cargo.crates` block above this as the required command. Yes, it may change over versions, but that can be found when we do updates.

This would match what @essandess suggests and make things *much* simpler overall for others to help keep this up to date.

Reply:
Creating the `cargo.crates` value for rust and rust-bootstrap is a bit of a pain.
There may be an easier way, but I have to resort to running `cargo2port` in multiple directories (listed in the comments) and directing the outputs to a file.
I manually tweak the file and remove duplicates via `sort -u`.
Perhaps we should wait on refining the comments until the pending improvements to `cargo2port` are available.

Reply:
@halostatue Where is your cargo2port PR?

Reply:
https://github.com/l2dy/cargo2port/pull/4

Reply:
The cargo2port script in the support folder *does not work at all* with modern Rust. I know that @herbygillot also has a Python script that does something similar, but I had switched to the Rust version and found it quite useful.

Reply:
Hrm. Why do we maintain this outside the MacPorts project? (Makes it harder for people to discover.)

Reply:
It’s ld2y's personal project as far as I can tell. It is available within MacPorts, which is how I installed it originally.